libs/backend/src/system_call/export_logs_to_usb.ts

```diff
@@ -30,6 +37,30 @@ export type LogsResultType = Result<
   | 'error-filtering-failed'
 >;
 
+function convertFileToCdf(
+  logger: Logger,
+  inputPath: string,
+  outputPath: string,
+  machineId: string,
+  codeVersion: string
+): Promise<void> {
+  const { promise, reject, resolve } = deferred<void>();
+
+  convertVxLogToCdf(
+    (eventId, message, disposition) => {
+      void logger.logAsCurrentRole(eventId, { message, disposition });
+    },
+    logger.getSource(),
+    machineId,
+    codeVersion,
+    inputPath,
+    outputPath,
+    (error) => (error ? reject(error) : resolve())
+  );
+
+  return promise;
```

Comment:
Nit: I'm not necessarily opposed to `deferred` but I think you could also just use the built-in `Promise` constructor:

```ts
return new Promise((resolve, reject) => 
   convertVxLogToCdf(
    (eventId, message, disposition) => {
      void logger.logAsCurrentRole(eventId, { message, disposition });
    },
    logger.getSource(),
    machineId,
    codeVersion,
    inputPath,
    outputPath,
    (error) => (error ? reject(error) : resolve())
  )
)
```

Related: Is there a reason `convertVxLogToCdf` takes a callback vs returning a promise?

Reply:
Yeah, was thinking the same - I'm guessing the callback pattern was easier to implement on the Rust side, but considering switching it over to promise along with the change to support compressed files